lib/asciidoctor/attribute_list.rb

```diff
@@ -170,11 +169,11 @@ def parse_attribute index, positional_attrs
         end
       end
     else
+      index = @attributes.select {|key| key.is_a? Numeric }.size
```

Comment:
Not the most efficient implementation but `@attributes` is unlikely to exceed 10 items and it removes the `index` parameter from the `parse_attribute` method.

Alternatively, we could return an extra boolean to indicate that we've added a positional attribute and in that case increment the index.